app/Http/Controllers/Rpc/WithdrawalController.php

```diff
@@ -21,11 +21,93 @@
 namespace Adshares\Adserver\Http\Controllers\Rpc;
 
 use Adshares\Adserver\Http\Controllers\Controller;
+use Adshares\Adserver\Utilities\AdsUtils;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Validator;
 
 class WithdrawalController extends Controller
 {
-    public function info()
+    /**
+     * Checks, if user has enough funds.
+     * @param $amount int transfer total amount
+     * @return bool true if has enough, false otherwise
+     */
+    public function hasUserEnoughFunds(int $amount): bool
     {
-        return self::json(json_decode('{"amount":10,"fee":9,"total":10}'));
+        // TODO check user account balance
+        $balance = 4000000000000000000;
+        return $amount <= $balance;
+    }
+
+    public function calculateWithdrawal(Request $request)
+    {
+        Validator::make($request->all(), [
+            'amount' => ['required', 'integer', 'min:1'],
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "amount" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "required" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)